web/pdf_page_view.js

```diff
@@ -468,19 +472,15 @@ class PDFPageView {
     if (!this.textLayer) {
       return;
     }
-    this.structTreeLayer ||= new StructTreeLayerBuilder();
 
-    const tree = await (!this.structTreeLayer.renderingDone
-      ? this.pdfPage.getStructTree()
-      : null);
-    const treeDom = this.structTreeLayer?.render(tree);
-    if (treeDom) {
+    const treeDom = await this.structTreeLayer.render();
+    if (treeDom && this.canvas && treeDom.parentNode !== this.canvas) {
       // Pause translation when inserting the structTree in the DOM.
       this.l10n.pause();
-      this.canvas?.append(treeDom);
+      this.canvas.append(treeDom);
       this.l10n.resume();
     }
-    this.structTreeLayer?.show();
+    this.structTreeLayer.show();
```

Comment:
Please revert this change, to prevent any theoretical issues if the structTreeLayer was removed *while* the render-promise resolved.
```suggestion
    this.structTreeLayer?.show();
```